src/components/manage/AdminUser/LrAddOrEditUser.vue

```diff
@@ -0,0 +1,323 @@
+            </v-row>
+          </v-form>
+        </v-container>
+      </v-card-text>
+      <v-divider />
+      <v-card-actions>
+        <v-container>
+          <v-row>
+            <v-col cols="12" class="btn-container">
+              <v-btn
+                class="white--text"
+                color="csmprimary"
+                @click="proceedOperation()"
+                :disabled="!isFormValid"
+                >{{ formLabelItems.ctaBtnLabel }}</v-btn
+              >
+              <v-btn
+                class="white--text"
+                color="csmborder"
+                @click="cancelOperation()"
+                >Cancel</v-btn
+              >
+            </v-col>
+          </v-row>
+        </v-container>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, Watch } from "vue-property-decorator";
+import SgtTooltipIcon from "../../../lib/components/SgtTooltipIcon/SgtTooltipIcon.vue";
+import {
+  usernameRegex,
+  emailRegex,
+  passwordRegex,
+} from "../../../utils/RegexHelpers";
+
+@Component({
+  name: "LrAddOrEditUser",
+  components: {
+    SgtTooltipIcon,
+  },
+})
+export default class LrAddOrEditUser extends Vue {
+  @Prop({ required: true }) value: boolean;
+  @Prop({ required: false }) userData: {
+    username: string;
+    role: "admin" | "manage" | "monitor";
+    email: string;
+  };
+
+  public isFormValid = false;
+  public userDetails = {
+    username: "",
+    role: "",
+    email: "",
+    currentPassword: "",
+    password: "",
+    confirmPassword: "",
+  };
+
+  public validationRules = {
+    username: [
+      (val: string) => (val || "").length > 0 || "Username is required",
+      (val: string) => usernameRegex.test(val) || "Invalid username",
+    ],
+    email: [
+      (val: string) => (val || "").length > 0 || "Email is required",
+      (val: string) => emailRegex.test(val) || "Invalid email",
+    ],
+    password: [
+      (val: string) => (val || "").length > 0 || "Password is required",
+      (val: string) => passwordRegex.test(val) || "Invalid password",
+    ],
+    confirmPassword: [
+      (val: string) => (val || "").length > 0 || "Password is required",
+      (val: string) =>
+        val === this.userDetails.password || "Passwords don't match",
+    ],
+  };
+
+  public formLabels = {
+    formTitle: "Add New User",
+    passwordLabel: "Password",
+    ctaBtnLabel: "Create",
+  };
+
+  get formLabelItems() {
+    if (this.userData) {
+      return {
+        formTitle: "Edit user account",
+        passwordLabel: "New Password",
+        ctaBtnLabel: "Update",
+      };
+    }
+    return this.formLabels;
+  }
+
+  public mounted() {
+    if (this.userData) {
+      this.userDetails.username = this.userData.username;
+      this.userDetails.email = this.userData.email;
+      this.userDetails.role = this.userData.role;
+    }
+  }
+
+  private strEqualityCaseInsensitive(first: string, second: string) {
```

Comment:
remove if not in use

Reply:
This function may require when we integrate the component with the API.

Reply:
Move this to common util function inside lib/services

Reply:
Done